httpclient/httpclient.go

```diff
@@ -335,6 +336,15 @@ func Timeout(timeout time.Duration) func(*Request) {
 	}
 }
 
+// MaxElapsedTime sets the overall timeout of a retryable request,
+// This is like setting the timeout field on the http client but only applies to this request.
+// To set a per-request timeout, use Timeout
```

Comment:
perhaps clarify that this fully overrides the per client timeout.

 (rather than the max or min of either)